app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -19,7 +19,8 @@ def google_oauth2
   def log_in_user_via_google
     flash[:notice] = I18n.t 'devise.omniauth_callbacks.success',
                             kind: 'Google'
-    sign_in_and_redirect @user, event: :authentication
+    sign_in @user, event: :authentication
```

Comment:
`sign_in_and_redirect` doesn't fire off devise callbacks for some extremely obnoxious reason! So I had to make an edit to get google oauth to bump `last_sign_in_at`, `last_sign_in_ip`, etc.